scipy/sparse/linalg/_special_sparse_arrays.py

```diff
@@ -0,0 +1,394 @@
+    array([[[0],
+            [1],
+            [2]],
+    <BLANKLINE>
+           [[3],
+            [4],
+            [5]]])
+
+    Each of the boundary conditions ``'Dirichlet'``, ``'Periodic'``, and
+    ``'Neumann'`` is illustrated separately.
+
+    >>> b_c = 'Dirichlet'
+    >>> lap = Laplacian_Nd(shape=shape, boundary_conditions=b_c)
+    >>> lap.tosparse()
+    <6x6 sparse matrix of type '<class 'numpy.int32'>'
+        with 20 stored elements in Compressed Sparse Row format>
+    >>> lap.toarray()
+    array([[-4,  1,  0,  1,  0,  0],
+           [ 1, -4,  1,  0,  1,  0],
+           [ 0,  1, -4,  0,  0,  1],
+           [ 1,  0,  0, -4,  1,  0],
+           [ 0,  1,  0,  1, -4,  1],
+           [ 0,  0,  1,  0,  1, -4]])
+    >>> np.array_equal(lap(np.eye(n)), lap.toarray())
+    True
+    >>> np.array_equal(lap.tosparse().toarray(), lap.toarray())
+    True
+    >>> lap.eigenvalues
+    array([-6.41421356, -5.        , -4.41421356, -3.58578644, -3.        ,
+           -1.58578644])
+    >>> eigvals = np.sort(eigh(lap.toarray(), eigvals_only=True))
+    >>> np.allclose(lap.eigenvalues, eigvals)
+    True
+    >>> b_c = 'Periodic'
+    >>> lap = Laplacian_Nd(shape=shape, boundary_conditions=b_c)
+    >>> lap.tosparse()
+    <6x6 sparse matrix of type '<class 'numpy.int32'>'
+        with 24 stored elements in Compressed Sparse Row format>
+    >>> lap.toarray()
+        array([[-4,  1,  1,  2,  0,  0],
+               [ 1, -4,  1,  0,  2,  0],
+               [ 1,  1, -4,  0,  0,  2],
+               [ 2,  0,  0, -4,  1,  1],
+               [ 0,  2,  0,  1, -4,  1],
+               [ 0,  0,  2,  1,  1, -4]])
+    >>> np.array_equal(lap(np.eye(n)), lap.toarray())
+    True
+    >>> np.array_equal(lap.tosparse().toarray(), lap.toarray())
+    True
+    >>> lap.eigenvalues
+    array([-7., -7., -4., -3., -3.,  0.])
+    >>> eigvals = np.sort(eigh(lap.toarray(), eigvals_only=True))
+    >>> np.allclose(lap.eigenvalues, eigvals)
+    True
+    >>> b_c = 'Neumann'
+    >>> lap = Laplacian_Nd(shape=shape, boundary_conditions=b_c)
+    >>> lap.sparse
+    <6x6 sparse matrix of type '<class 'numpy.int32'>'
+        with 20 stored elements in Compressed Sparse Row format>
+    >>> lap.toarray()
+    array([[-2.,  1.,  0.,  1.,  0.,  0.],
+           [ 1., -3.,  1.,  0.,  1.,  0.],
+           [ 0.,  1., -2.,  0.,  0.,  1.],
+           [ 1.,  0.,  0., -2.,  1.,  0.],
+           [ 0.,  1.,  0.,  1., -3.,  1.],
+           [ 0.,  0.,  1.,  0.,  1., -2.]])
+    >>> np.array_equal(lap(np.eye(n)), lap.toarray())
+    True
+    >>> np.array_equal(lap.tosparse().toarray(), lap.toarray())
+    True
+    >>> lap.eigenvalues
+    array([-5., -3., -3., -2., -1.,  0.])
+    >>> eigvals = np.sort(eigh(lap.toarray(), eigvals_only=True))
+    >>> np.allclose(lap.eigenvalues, eigvals)
+    True
+
+    """
+
+    def __init__(self, shape, *, boundary_conditions, dtype=np.int32) -> None:
+        self.dims = shape
+        self.bc = boundary_conditions
```

Comment:
Or perhaps even better (than making `self.bc` a property to avoid user assignments): just make the member private by calling it `_bc`. If we want to keep the member public, I'd prefer to name it `boundary_conditions`, like the kwarg of the class constructor.

Reply:
```suggestion
        self._bc = boundary_conditions
```

Reply:
done